### DISCUSSION TITLE
Help with supplemental figures in Typst

### DISCUSSION THREAD

**Discussion by mps9506** [quarto-dev/quarto-cli#9413] [Q&A]
### Description

I'm attempting to use the custom float cross reference types to create supplementary figure and table captions and cross references (eg. Fig S1, Table S1) with the Typst output. Here is what I am currently doing:

    ---
    title: "MWE"
    format: typst
    
    crossref:
      custom:
        - kind: float
          key: suppfig
          reference-prefix: Figure S
          caption-location: bottom
          space-before-numbering: false
    ---
    
    :::{#suppfig-test}
    ```{r}
    #| echo: false
    plot(rnorm(50),rnorm(50))
    ```
    test caption
    ::::
 
This nearly works but I get an undesired space between the prefix and the number:
![image](https://github.com/quarto-dev/quarto-cli/assets/11282246/c18d18cf-b961-49fa-8962-be58f93e083b)

Looking at the .typ output, the figure block looks like:
```typst
#figure([
#block[
#block[
#box(width: 396.0pt, image("mew_files/figure-typst/unnamed-chunk-1-1.svg"))

]
]
], caption: figure.caption(
position: bottom, 
[
test caption
]), 
kind: "quarto-float-suppfig", 
supplement: "Figure S", 
numbering: "1", 
)
<suppfig-test>
```

There is no obvious way to control spacing using the custom float. I'm wondering if I'm missing a way to specify the `numbering` parameter via Quarto instead (https://typst.app/docs/reference/model/numbering/)? Something like `numbering: "S1"`

Here is my session info:

```
R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8    

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.3.1 tools_4.3.1    renv_1.0.3  
```

output from `quarto --version`: `1.4.553`

**Comment by cscheid** [comment]
We haven't implemented support for `space-before-numbering` in typst yet.

> Something like `numbering: "S1"`

That's not going to work; what needs to happen is more intricate, and involves redefining the function that takes `supplement` and `numbering` and produces the caption content.

**Reply by mcanouil** [reply]
- https://github.com/quarto-dev/quarto-cli/issues/9416
